src/taskHandler.ts

```diff
@@ -37,23 +36,23 @@ export class TaskHandler {
     );
   }
 
-  public async waitForTask<T>(taskType: string): Promise<ITaskResponse<T>> {
+  public async waitForTask<T>(taskType: string, jobType: string): Promise<ITaskResponse<T>> {
     let task: ITaskResponse<T> | null;
     this.logger.info({
-      jobType: this.jobType,
+      jobType,
       taskType,
-      msg: `waitForTask jobType=${this.jobType}, taskType=${taskType}`,
+      msg: `waitForTask jobType=${jobType}, taskType=${taskType}`,
     });
     do {
-      task = await this.dequeue(taskType);
+      task = await this.dequeue(taskType, jobType);
       await new Promise((resolve) => setTimeout(resolve, this.dequeueIntervalMs));
     } while (!task);
     return task;
   }
 
-  public async dequeue<T>(taskType: string): Promise<ITaskResponse<T> | null> {
+  public async dequeue<T>(taskType: string, jobType: string): Promise<ITaskResponse<T> | null> {
```

Comment:
please put the jobType before the task type

Reply:
done